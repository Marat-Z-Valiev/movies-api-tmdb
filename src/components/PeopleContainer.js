import React from "react";
import {Link} from "react-router-dom";
import PropTypes from "prop-types";
import PeopleContainerStyled from "./styles/PeopleContainerStyled";
import Spinner from "./Spinner";
import noImage from "../images/no-image-available.jpg";

const PeopleContainer = ({people, isLoading}) => {
	return (
		<>
			{isLoading ? (
				<Spinner />
			) : (
				<PeopleContainerStyled>
					{Object.keys(people).map((key) => (
						<div key={people[key].id} className="person">
							<Link to={`/personId=${people[key].id}`}>
								{people[key].profile_path == null ? (
									<img
										className="no-image"
										src={noImage}
										alt="no image available"
									/>
								) : (
									<img
										src={`https://image.tmdb.org/t/p/w342/${people[key].profile_path}`}
										alt={people[key].name}
									/>
								)}
							</Link>
							<h2 className="name">{people[key].name}</h2>
						</div>
					))}
				</PeopleContainerStyled>
			)}
		</>
	);
};

PeopleContainer.propTypes = {
	people: PropTypes.array.isRequired,
	isLoading: PropTypes.bool,
};

export default PeopleContainer;
