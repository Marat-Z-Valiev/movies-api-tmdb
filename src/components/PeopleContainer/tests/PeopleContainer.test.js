import React from "react";
import {shallow, configure} from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import PeopleContainer from "../PeopleContainer";

configure({adapter: new Adapter()});

describe("PeopleContainer", () => {
	it("should render correctly", () => {
		const props = {
			people: [],
			isLoading: true,
		};
		const wrapper = shallow(<PeopleContainer {...props} />);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});
});
