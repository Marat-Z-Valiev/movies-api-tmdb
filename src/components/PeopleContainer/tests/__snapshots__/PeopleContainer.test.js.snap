// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleContainer should render correctly 1`] = `
<Fragment>
  <Spinner />
</Fragment>
`;
